File: Byte&Buy.Backend/src/ByteBuy.Core/DTO/Public/Checkout/SellerGroup.cs
using ByteBuy.Core.DTO.Public.Money;

namespace ByteBuy.Core.DTO.Public.Checkout;

public record SellerGroup(
    Guid SellerId,
    string SellerDisplayName,
    string SellerEmail,
    MoneyDto ItemsWorth,
    IReadOnlyCollection<CheckoutItem> CheckoutItems);

File: Byte&Buy.AngularSPA/src/app/core/dto/checkout/seller-group.ts
import { Guid } from "guid-typescript";
import { MoneyDto } from "../common/money-dto";
import { DeliveryOptionsResponse } from "../delivery/delivery-options-response";
import { RentCheckoutItem } from "./rent-checkout-item";
import { SaleCheckoutItem } from "./sale-checkout-item";

export type CheckoutItem = SaleCheckoutItem | RentCheckoutItem;

export interface SellerGroup {
  sellerId: Guid;
  sellerDisplayName: string;
  sellerEmail: string;
  itemsWorth: MoneyDto;
  checkoutItems: CheckoutItem[];
  deliveryOptions: DeliveryOptionsResponse
}